File: scala/src/main/scala/tddmicroexercises/textconvertor/HtmlTextConverter.scala
package tddmicroexercises.textconvertor

import java.io.{FileReader, BufferedReader}


class HtmlTextConverter(val fullFilenameWithPath : String) {

  def convertToHtml() : String  = {

    val reader = new BufferedReader(new FileReader(fullFilenameWithPath))

    var line = reader.readLine()
    var html = ""
    while (line != null)
    {
      html += StringEscapeUtils.escapeHtml(line)
      html += "<br />"
      line = reader.readLine()
    }
    return html

  }

}

File: Cpp/Source/TextConverter/source/HtmlTextConverter.cpp

#include "../include/HtmlTextConverter.h"
#include "../include/StringEscapeUtils.h"

HtmlTextConverter::HtmlTextConverter(std::string const& fullFilenameWithPath) : m_fullFilenameWithPath(fullFilenameWithPath)
{}

std::string HtmlTextConverter::convertToHtml() 
{
    std::ifstream reader(m_fullFilenameWithPath);

    std::string line;
    std::string html;
    while (std::getline(reader,line))
    {
        html += StringEscapeUtils::escapeHtml(line);
        html += "<br />";
    }
    return html;
}

std::string HtmlTextConverter::getFilename() 
{
    return m_fullFilenameWithPath;
}
